crates/orchestrator/src/lib.rs

```diff
@@ -416,8 +436,33 @@ where
     }
 
     // Assumes nodes do not post 'ready' twice
-    // TODO ED Add a map to verify which nodes have posted they're ready
-    fn post_ready(&mut self) -> Result<(), ServerError> {
+    fn post_ready(&mut self, pubkey: &mut Vec<u8>) -> Result<(), ServerError> {
+        // Deserialize the public key
+        let staked_pubkey = PeerConfig::<KEY>::from_bytes(pubkey).unwrap();
+
+        // Is this node allowed to connect?
+        if !self
+            .config
+            .public_keys
+            .contains(&staked_pubkey.stake_table_entry.public_key())
+        {
+            return Err(ServerError {
+                status: tide_disco::StatusCode::FORBIDDEN,
+                message: "You are unauthorized to register with the orchestrator".to_string(),
+            });
+        }
+
+        // Have they already connected?
+        if self.ready_posted.contains(pubkey) {
```

Comment:
can ready_posted be a hashset and we do something like:
```suggestion
        if !self.ready_posted.insert(pubkey) {
        ...
        }
```

then we also wouldnt need to push onto the vector after this

Reply:
Done